test/unit/cmap/connection.test.ts

```diff
@@ -486,4 +593,58 @@ describe('new Connection()', function () {
       });
     });
   });
+
+  describe('destroy()', () => {
+    let connection: sinon.SinonSpiedInstance<Connection>;
+    let clock: sinon.SinonFakeTimers;
+    let timerSandbox: sinon.SinonFakeTimers;
+    let driverSocket: sinon.SinonSpiedInstance<FakeSocket>;
+    let messageStream: MessageStream;
+    beforeEach(() => {
+      timerSandbox = createTimerSandbox();
+      clock = sinon.useFakeTimers();
+
+      driverSocket = sinon.spy(new FakeSocket());
+      // @ts-expect-error: driverSocket does not fully satisfy the stream type, but that's okay
+      connection = sinon.spy(new Connection(driverSocket, connectionOptionsDefaults));
+      const messageStreamSymbol = getSymbolFrom(connection, 'messageStream');
+      messageStream = sinon.spy(connection[messageStreamSymbol]);
+    });
+
+    afterEach(() => {
+      timerSandbox.restore();
+      clock.restore();
+    });
+
+    it('ends the tcp socket and destroys the messageStream', () => {
+      connection.destroy({ force: false });
+      clock.tick(1);
+      expect(messageStream.destroy).to.have.been.calledOnce;
+      expect(driverSocket.end).to.have.been.calledOnce;
+    });
+
+    it('calls stream.end exactly once when destroy is called multiple times', () => {
+      connection.destroy({ force: false });
+      connection.destroy({ force: false });
+      connection.destroy({ force: false });
+      connection.destroy({ force: false });
+      clock.tick(1);
+      expect(driverSocket.destroy).to.not.have.been.called;
+      expect(driverSocket.end).to.have.been.calledOnce;
+    });
+
+    it('does not call stream.end if options.force == true', () => {
```

Comment:
While it's nice to know what `options.force == true` doesn't do, it would be even more useful to know what it does do.

Can we split up the cases outlined here into two context blocks, one with force set to true and another with force set to false and ensure that the test cases fully capture the intended behavior for each option setting?